lib/vdsm/storage/blockSD.py

```diff
@@ -1071,11 +1071,10 @@ def create(cls, sdUUID, domainName, domClass, vgUUID, storageType,
 
         if set((STORAGE_UNREADY_DOMAIN_TAG,)) != set(vg.tags):
             raise se.VolumeGroupHasDomainTag(vgUUID)
-        try:
-            lvm.getAllLVs(vgName)
+
+        if lvm.getAllLVs(vgName):
+            # There are already LVs in the VG
             raise se.StorageDomainNotEmpty(vgUUID)
-        except se.LogicalVolumeDoesNotExistError:
-            pass
```

Comment:
Much nicer like this - but do we have a case when lvm.getAllLVs(vgName) raises
and we need to handle here?

Previously we assumed that this error means the vg is empty. I wonder if we could
get this error when the vg was not empty, but we cannot access storage.

The new code seems better, in case of errors it will raise and fail the current flow.

Reply:
I don't think so. We reload LVs through an LVM with silent errors, and only print a warning in a reload failure for an Unreadable volume. Other than that, we do not raise anymore for the `AllLVs` getter.

I think the new code expresses better the fact that we expect an empty VG (without LVs). Maybe I'll add a `> 0` to the condition to make it more explicit.